core/src/main/java/org/apache/iceberg/IncrementalDataTableScan.java

```diff
@@ -106,7 +106,7 @@ protected TableScan newRefinedScan(TableOperations ops, Table table, Schema sche
     return new IncrementalDataTableScan(ops, table, schema, context);
   }
 
-  private static List<Snapshot> snapshotsWithin(Table table, long fromSnapshotId, long toSnapshotId) {
+  public static List<Snapshot> snapshotsWithin(Table table, long fromSnapshotId, long toSnapshotId) {
```

Comment:
We only really need this method to be public so we can determine the snapshots to be used in the scan

Reply:
Should we move this to `SnapshotUtil` then?